radicle-node/src/service/tracking.rs

```diff
@@ -14,43 +14,73 @@ pub use store::Error;
 #[derive(Debug)]
 pub struct Config {
     /// Default policy, if a policy for a specific node or repository was not found.
-    default: Policy,
+    policy: Policy,
+    /// Default scope, if a scope for a specific repository was not found.
+    scope: Scope,
     /// Underlying configuration store.
     store: store::Config,
 }
 
 impl Config {
     /// Create a new tracking configuration.
-    pub fn new(default: Policy, store: store::Config) -> Self {
-        Self { default, store }
+    pub fn new(policy: Policy, scope: Scope, store: store::Config) -> Self {
+        Self {
+            policy,
+            scope,
+            store,
+        }
+    }
+
+    /// Determine whether the `id` should be fetched.
+    ///
+    /// If an entry for this `id` exists and its policy is
+    /// [`Policy::Track`], then this will return `true`.
+    ///
+    /// Otherwise, if the `id` does not exist, and the default
+    /// [`Self::scope`] is [`Scope::All`] it returns `true`.
+    ///
+    /// It returns `false` if neither of these is the case.
+    pub fn should_fetch(&self, id: &Id) -> Result<bool, Error> {
+        Ok(match self.store.repo_entry(id)? {
+            Some(entry) => entry.policy == Policy::Track,
+            None => matches!(self.scope, Scope::All),
+        })
     }
 
     /// Check if a repository is tracked.
     pub fn is_repo_tracked(&self, id: &Id) -> Result<bool, Error> {
-        self.repo_policy(id).map(|policy| policy == Policy::Track)
+        self.store.is_repo_tracked(id)
     }
 
     /// Check if a node is tracked.
     pub fn is_node_tracked(&self, id: &NodeId) -> Result<bool, Error> {
-        self.node_policy(id).map(|policy| policy == Policy::Track)
+        self.store.is_node_tracked(id)
     }
```

Comment:
Hmmmmmm these functions no longer make use of the default policies in case something isn't in the db.. is that intentional?

Reply:
So, your tracking policy is a combination of the entries in the tracking db, plus your default policy. So the idea is that if there's no entry in the db, we use whatever your default, or "fallback" policy is.

Example for repos:

```
* default=track, db=None    -> track
* default=block, db=None    -> block
* default=track, db=block     -> block
* default=block, db=track     -> track 
```

That's for repositories, now for the cases where we track, we can additional check the tracking scope, to see if a specific remote should be fetched:

```
* repo=tracked, scope=all         -> fetch the remote
* repo=tracked, scope=trusted  -> check if node is tracked in db
```

Reply:
Let me digest and play with this to understand a bit better :)

Reply:
Ok, I changed those functions back. I think its use in [`is_tracking`](https://github.com/radicle-dev/heartwood/blob/master/radicle-node/src/service.rs) is a bit of a lie though, in a way, but not a big deal for this change set. I just found it confusing that it would say `true` if I didn't actually track anything -- which was causing me issues when trying to figure out test failures. However, with this current set up proposed here now everything works at each commit :)

Reply:
Yeah, I agree with that actually. Happy to try to make the API clearer by distinguishing actual tracking with a "Track" policy.